ensight_lib/include/ensightdef.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2016 Fraunhofer ITWM
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+
+#ifndef ENSIGHTDEF_H
+#define ENSIGHTDEF_H
+
+/**
+ * The namespace Ensight summarizes the structural key words of the EnSight Gold format.
+ */
+namespace Ensight {
+
+/**
+ * @brief The Section enum contains the five sections used in the EnSight Gold case format:
+ * (FORMAT, GEOMETRY, VARIABLE, TIME, FILE).
+ */
+enum Section {SectionUnknown=-1, Format=0, Geometry=1, Variable=2, Time=3, File=4};
```

Comment:
Prefer 'enum class'

Also, a common practice is to add an extra item called 'LAST' or similar so that you are "guaranteed" to stay in sync with your self (i.e. if you added another Section type here but don't update line 38, that's bad news).

E.g.
enum Section {SectionUnknown=-1, Format=0, Geometry=1, Variable=2, Time=3, File=4, numSections = 5};

forces things to stay in sync a little bit more.

Reply:
The enum's are already in namespace Ensight, so I'd like to avoid introducing another scope level with enum class.
I've added end markers as you suggested.